cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -2839,6 +2945,138 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
             }
         }
 
+        Opcode::Splat => {
+            let ty = ty.unwrap();
+            assert_eq!(ty.bits(), 128);
+            let src_ty = ctx.input_ty(insn, 0);
+            assert!(src_ty.bits() < 128);
+            let src = input_to_reg_mem(ctx, inputs[0]);
+            let dst = get_output_reg(ctx, outputs[0]);
+
+            fn emit_insert_lane<C: LowerCtx<I = Inst>>(
```

Comment:
Let's pull this out into a toplevel helper; I'd rather keep the main `match` from exploding in size with nested functions :-)

Reply:
Good point; that was the intent but I forgot to 😄.